cdi-core/src/main/java/com/linkedin/cdi/extractor/MultistageExtractor.java

```diff
@@ -976,9 +976,16 @@ protected void readPayloads(State state) {
     JsonArray records = new JsonArray();
     for (JsonElement entry : payloads) {
       try {
-        records.addAll(new HdfsReader(state).readSecondary(entry.getAsJsonObject()));
-        extractorKeys.setPayloads(records);
-        payloadIterator = records.iterator();
+        JsonObject entryObject = entry.getAsJsonObject();
+        if (shouldReadAsBinary(entryObject)) {
```

Comment:
Please confirm whether you tested multiple file uploads in the testing section.

Reply:
Just did